google/resource_compute_disk.go

```diff
@@ -21,106 +35,372 @@ var (
+						},
+						"sha256": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
 			},
-
-			"self_link": &schema.Schema{
-				Type:     schema.TypeString,
-				Computed: true,
+			"source_image_encryption_key": {
+				Type:     schema.TypeList,
+				Optional: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"raw_key": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+						},
+						"sha256": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
 			},
-
-			"snapshot": &schema.Schema{
+			"snapshot": {
 				Type:             schema.TypeString,
 				Optional:         true,
 				ForceNew:         true,
 				DiffSuppressFunc: linkDiffSuppress,
 			},
-
-			"type": &schema.Schema{
-				Type:     schema.TypeString,
+			"source_snapshot_encryption_key": {
+				Type:     schema.TypeList,
 				Optional: true,
-				Default:  "pd-standard",
 				ForceNew: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"raw_key": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+						},
+						"sha256": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
 			},
-
-			"users": &schema.Schema{
+			"creation_timestamp": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"last_attach_timestamp": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"last_detach_timestamp": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"source_image_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"source_snapshot_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"users": {
 				Type:     schema.TypeList,
 				Computed: true,
-				Elem:     &schema.Schema{Type: schema.TypeString},
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"disk_encryption_key_raw": &schema.Schema{
+				Type:       schema.TypeString,
+				Optional:   true,
+				ForceNew:   true,
+				Sensitive:  true,
+				Deprecated: "Use disk_encryption_key.raw_key instead.",
 			},
 
-			"labels": &schema.Schema{
-				Type:     schema.TypeMap,
-				Optional: true,
-				Elem:     &schema.Schema{Type: schema.TypeString},
+			"disk_encryption_key_sha256": &schema.Schema{
+				Type:       schema.TypeString,
+				Computed:   true,
+				Deprecated: "Use disk_encryption_key.sha256 instead.",
 			},
 
 			"label_fingerprint": &schema.Schema{
 				Type:     schema.TypeString,
 				Computed: true,
 			},
+			"project": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"self_link": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
 		},
-		CustomizeDiff: customdiff.All(
-			customdiff.ForceNewIfChange("size", isDiskShrinkage)),
 	}
```

Comment:
this isDiskShrinkage customizediff doesn't seem to have made it in

Reply:
Done - I forgot, I added that after I started working on Disk generation.  :man_facepalming: